File: src/main/java/com/jolvera/jbud/domain/abstracts/RecurringExpense.java
package com.jolvera.jbud.domain.abstracts;

import com.jolvera.jbud.domain.enums.RecurrencePattern;
import com.jolvera.jbud.domain.enums.TransactionType;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.UUID;

public abstract class RecurringExpense {
    private UUID id;
    private String description;
    private BigDecimal amount;
    private LocalDate startDate;
    private boolean active;
    private TransactionType type;
    private RecurrencePattern recurrencePattern;


    public LocalDate getNextOccurrenceDate() {
        throw new UnsupportedOperationException("Not implemented yet");
    }
}

File: src/main/kotlin/com/jolvera/jbud/domain/abstracts/RecurringExpense.kt
package com.jolvera.jbud.domain.abstracts

import com.jolvera.jbud.domain.RecurrancePattern
import domain.enums.TransactionType
import java.time.Instant
import java.util.*

class RecurringExpense(
    override val id: UUID = UUID.randomUUID(),
    override val description: String,
    override val amount: Double,
    override val startDate: Instant,
    override val type: TransactionType,
    override val pattern: RecurrancePattern
) : RecurringTransaction() {
    private val payee: String? = null
    private val paymentMethod: String? = null
    private val notes: String? = null
    private val autoRenewable: Boolean? = null
}